### ISSUE TITLE
Subscribing without takeUntil is forbidden (rxjs-prefer-angular-takeuntil) when take(1) is added

### ISSUE THREAD

**Issue by Abasz** [cartant/rxjs-tslint-rules#114]
Consider the follwoing code:

```javascript
snackRef.afterDismissed().pipe(take(1)).subscribe(() => {
   // do something
});
```

rxjs-tslint-rules shows an error that _"Subscribing without takeUntil is forbidden"_ which is absolutely correct in the sense that there is no takeUntil.

But I think since here unsubscribing is done automatically due to take(1) after one emission, the prefer-angular-takeuntil rule should not show error.

Is this intentional or am I setting up something wrong here?

Thanks in advance

**Comment by cartant** [comment]
Whether or not unsubscription happens 'automatically' - i.e. when/if the source completes - isn't necessarily the issue. `takeUntil` will explicitly unsubscribe when the notifier signals and that could effect a behaviour that `take` would not - e.g. cancelling a pending HTTP request.

If you have circumstances in which `takeUntil` is not needed, IMO, you should disable the rule on a case-by-case basis using the [linter-supported configuration comments](https://palantir.github.io/tslint/usage/rule-flags/). I have no intention of making the rule more complicated than it already is.

Whether or not you use the rule at all is up to you. A large number of the rules in this package were added because developers requested them.
